packages/ngtools/webpack/src/resource_loader.ts

```diff
@@ -110,17 +111,23 @@ export class WebpackResourceLoader {
       throw new Error('WebpackResourceLoader cannot be used without parentCompilation');
     }
 
-    // Create a special URL for reading the resource from memory
-    const entry =
-      filePath ||
-      (resourceType
-        ? `${containingFile}-${this.outputPathCounter}.${fileExtension}!=!${this.inlineDataLoaderPath}!${containingFile}`
-        : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-          `angular-resource:${resourceType},${createHash('md5').update(data!).digest('hex')}`);
+    const getEntry = (): string => {
```

Comment:
I like how you put this in a separate function to take advantage of `return`, I think it's the most elegant way of handling this kind of logic.